tests-api/app/use_cases/calculate_score.rb

```diff
@@ -31,23 +31,28 @@ def calculate_results
     @correct_questions = 0
 
     answers.map do |answer|
+      correct_choices = correct_choices(answer[:question_uuid])
       correct = answered_correctly?(
-        answer[:question_uuid],
         answer[:selected_choices],
+        correct_choices,
       )
 
-      @correct_questions += 1 if correct
+      @correct_questions += 1
 
       {
         question_uuid: answer[:question_uuid],
+        correct_answers: correct_choices,
         correct: correct,
```

Comment:
maybe rename `correct` to avoid confusion with `correct_answers` ? `answered_correctly` maybe?